### PR TITLE
A few nullable changes

### PR DESCRIPTION
**Pull request by MihaZupan**
cc: @iamcarbon

### CODE REVIEW

File: src/Markdig.Tests/TestReferralLinks.cs [hunk 1]
```diff
@@ -37,20 +37,6 @@ public void TestLinksWithCustomRel(string[] rels, string expected)
...
-            var markdown = "[world](http://example.com)";
-
-            var pipeline = new MarkdownPipelineBuilder()
-                .UseReferralLinks(null)
-                .Build();
-            var html = Markdown.ToHtml(markdown, pipeline);
-
-            Assert.That(html, !Contains.Substring("rel="));
-        }
-
```
Comment (MihaZupan) [inline src/Markdig.Tests/TestReferralLinks.cs:53]: I don't know if I'm missing something, but I don't understand this test.

If you don't want the `rel` being output... don't add the extension?

File: src/Markdig/Renderers/Html/Inlines/AutolinkInlineRenderer.cs [hunk 2]
```diff
@@ -25,14 +25,22 @@ public bool AutoRelNoFollow
...
+                const string NoFollow = "nofollow";
+
+                if (value)
                 {
-                    Rel = string.IsNullOrEmpty(Rel) ? rel : Rel + $" {rel}";
+                    if (string.IsNullOrEmpty(Rel))
+                    {
+                        Rel = NoFollow;
+                    }
+                    else if (!Rel!.Contains(NoFollow))
```
Comment (MihaZupan) [inline src/Markdig/Renderers/Html/Inlines/AutolinkInlineRenderer.cs:36]: This `!` isn't needed on Core 3.1+, sadly older runtimes aren't smart enough to realize `string.IsNullOrEmpty` is a null check

Comment (iamcarbon) [reply src/Markdig/Renderers/Html/Inlines/AutolinkInlineRenderer.cs:36]: There's a NRE condition here that we're not handling correctly. We should rewrite this code to correctly handle a null Rel.

Comment (MihaZupan) [reply src/Markdig/Renderers/Html/Inlines/AutolinkInlineRenderer.cs:36]: It does handle it via `string.IsNullOrEmpty`

Comment (iamcarbon) [reply src/Markdig/Renderers/Html/Inlines/AutolinkInlineRenderer.cs:36]: Indeed. Missed the outer logic changes here. 👍

**Pull request merged**